spam/spam_tutorial.py

```diff
@@ -140,234 +140,400 @@
 
-correct = L_dev_array == Y_dev
-labeled = L_dev_array != ABSTAIN
-accuracy = (correct * labeled).sum() / labeled.sum()
-print(f"Accuracy: {accuracy * 100:.1f}%")
+# %%
+LFAnalysis(L=L_dev, lfs=lfs).lf_summary(Y=Y_dev)
 
 # %% [markdown]
-# Alternatively, you can use the provided `metric_score()` helper method, which allows you to specify a metric to calculate and certain classes to ignore (such as ABSTAIN).
+# So even these very simple rules do quite well!
+# Following our principle above, we might want to pick the `check` rule, since both have high precision and `check` has higher coverage.
+# But let's look at our data to be sure.
+#
+# The helper method `error_buckets()` groups examples by their predicted label and true label. For example, `buckets[(SPAM, HAM)]` contains the indices of data points that the LF labeled `SPAM` that actually belong to class `HAM`. This may give ideas for where the LF could be made more specific.
 
 # %%
-from snorkel.analysis.metrics import metric_score
+from snorkel.analysis.error_analysis import error_buckets
 
-# Calculate accuracy, ignore all examples for which the predicted label is ABSTAIN
-accuracy = metric_score(
-    golds=Y_dev, preds=L_dev_array, metric="accuracy", filter_dict={"preds": [ABSTAIN]}
-)
-print(f"Accuracy: {accuracy * 100:.1f} %")
+buckets = error_buckets(golds=Y_dev, preds=L_dev[:, 1])
+df_dev.iloc[buckets[(SPAM, HAM)]]
 
 # %% [markdown]
-# You can also use the helper class `LFAnalysis()` to report the following summary statistics for multiple LFs at once:
-# * **Polarity**: The set of labels this LF outputs
-# * **Coverage**: The fraction of the dataset the LF labels
-# * **Overlaps**: The fraction of the dataset where this LF and at least one other LF label
-# * **Conflicts**: The fraction of the dataset where this LF and at least one other LF label and disagree
-# * **Correct**: The number of data points this LF labels correctly (if gold labels are provided)
-# * **Incorrect**: The number of data points this LF labels incorrectly (if gold labels are provided)
-# * **Emp. Acc.**: The empirical accuracy of this LF (if gold labels are provided)
-#
-# Since we only have one LF, overlaps and conflicts will be 0.
+# `check` is still looking good, since the false positive is specific to only a few of the most popular videos on YouTube (ones with billions of views).
+# Now let's take a look at some places that `check` labeled `SPAM` on the training set to see if it matches our intuition or if we can identify some false positives.
 
 # %%
-from snorkel.labeling.analysis import LFAnalysis
+df_train.iloc[L_train[:, 1] == SPAM].sample(10, random_state=1)
 
-LFAnalysis(L=L_dev, lfs=lfs).lf_summary(Y=Y_dev)
+# %% [markdown]
+# No clear false positives here, but many look like they could be labeled by `check_out` as well.
+# Let's see where `check_out` abstained, but `check` labeled.
+
+# %%
+buckets = error_buckets(golds=L_train[:, 0], preds=L_train[:, 1])
```

Comment:
This is a little confusing, because (from the kwarg names) it looks like you are treating one LF as gold. Maybe the error buckets kwargs should just be renamed to 'left_preds' and 'right_preds'.

Reply:
agree about rename, will remove kwargs for now